lib/parse.coffee

```diff
@@ -27,7 +27,7 @@ exports.parse = (argv) ->
 	result = {}
 
 	result.options = _.mapValues options, (value) ->
-		if /^[\d\.]+$/.test(value)
+		if /^\d+(\.\d+)?$/.test(value)
```

Comment:
Although then you get the issue of matching an empty string, so I guess 
```coffee
if /^\d+$/.test(value)
	return parseInt(value)
if /^\d*\.\d+?$`.test(value)
	return parseFloat(value)
```
would be better